File: RunningText.py
#!/usr/bin/env python3
import sys


def main():
    CORE_TEXT = "SPACE AVAILABLE SPACE AVAILABLE "
    CORE_LEN = len(CORE_TEXT)

    pos = 0
    try:
        with open('/proc/uptime', 'r') as fh:
            ln = fh.readline()
            seconds_int = int(ln.split('.')[0])
            pos = seconds_int % CORE_LEN
    except Exception:
        pass

    long_txt = CORE_TEXT + CORE_TEXT

    moving_part = long_txt[pos:pos+32]

    rr = f'  {moving_part} '

    sys.stdout.write(rr)


if __name__ == "__main__":
    main()

File: RunningText.pl
#!/usr/bin/env perl
use strict;
use warnings;

my $CORE_TEXT = "SPACE AVAILABLE SPACE AVAILABLE ";
my $CORE_LEN  = length($CORE_TEXT);

my $pos = 0;
if (open(my $fh, '<', '/proc/uptime')) {
    my $ln = <$fh>;
    close($fh);
    if ($ln =~ /^(\d+)/) {
        $pos = $1 % $CORE_LEN;
    }
}

my $long_txt   = $CORE_TEXT . $CORE_TEXT;
my $moving_part = substr($long_txt, $pos, 32);
my $rr = "  " . $moving_part . " ";

print "$rr";
